axelrod/strategy_transformers.py

```diff
@@ -84,23 +92,18 @@ def __call__(self, PlayerClass):
             except KeyError:
                 pass
 
-            # Is the original strategy method a static method?
-            signature = inspect.signature(PlayerClass.strategy)
-            strategy_args = [p.name for p in signature.parameters.values()
-                    if p.kind == inspect.Parameter.POSITIONAL_OR_KEYWORD]
-            is_static = True
-            if len(strategy_args) > 1:
-                is_static = False
-
             # Define the new strategy method, wrapping the existing method
             # with `strategy_wrapper`
             def strategy(self, opponent):
-
-                if is_static:
-                    # static method
-                    proposed_action = PlayerClass.strategy(opponent)
+                if strategy_wrapper == dual_wrapper:
```

Comment:
(see the explanation of how the state change works first)

`wp` = wrapped_player
`op` = original_player

the calls inside `dual_wrapper` call `op.strategy` **on** `wp`.  this mutates the state of `wp` as if it was `op` .  if the `op.strategy` gets called here again, it will again mutate the state of `wp`.  this would be bad.  

ex of bad way (if uses proposed_action = PlayerClass.strategy(opponent):

player = DualCycler('CD') not yet played
expected_action = `C.flip()` = `D`
cycle now advanced to `D`
expected_player_state = next_action_to_be_returned: `D.flip()`

1) `wp.strategy` called
2) `proposed_action` calls `op.strategy`
3) returns `C` (which will be discarded)
4) mutates state of `wp` (cycle now advanced to 'D')
5) `dual_wrapper` called
6) `dual_wrapper` calls `op.strategy` 
7) returns `D`
8) mutates state of `wp`  (cycle now advanced to `C`, the wrong state)
9) return action.flip() (which is now `C`, the wrong action)

removing steps 2, 3, 4 makes `wp` return the correct action and stay in the correct state.